### PR TITLE
Show Role Audience of Assigned Roles of the Group

### PR DESCRIPTION
**Pull request by Shenali-SJ**
### Related Issues
- https://github.com/wso2/product-is/issues/17652

### Related PRs
- https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/509

### Revised UI
Screenshot 2023-11-18 at 22 05 15

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/groups/components/group-common-components/render-chip.tsx [hunk 1]
```diff
@@ -31,17 +30,21 @@ interface RenderChipInterface extends IdentifiableComponentInterface, ChipProps
      */
     setActiveOption: (option: RolesMemberInterface) => void;
     /**
-     * Primary text of the chip.
+     * Display name of the role
```
Comment (pavinduLakshan) [inline apps/console/src/features/groups/components/group-common-components/render-chip.tsx:33]: If we are updating the component properties specifically for displaying role information, let's also rename `RenderChip` and related interface names as well.

File: apps/console/src/features/groups/components/group-common-components/render-chip.tsx [hunk 2]
```diff
@@ -51,12 +54,13 @@ export const RenderChip: FunctionComponent<RenderChipInterface> = (
...
-        option,
-        activeOption
+        displayName,
+        audienceType,
+        audienceDisplay,
+        option
     } = props;
 
-    const [ popoverAnchorEl, setPopoverAnchorEl ] = useState<Element>(null);
+    const [ , setPopoverAnchorEl ] = useState<Element>(null);
```
Comment (pavinduLakshan) [inline apps/console/src/features/groups/components/group-common-components/render-chip.tsx:63]: Can't we remove this as well?

**Comment by Achintha444** [comment]
Can you add a screenshot on how it will display when assigned multiple org roles.

File: apps/console/src/features/groups/components/edit-group/edit-group-roles.tsx [hunk 3]
```diff
@@ -118,7 +118,10 @@ export const EditGroupRoles: FunctionComponent<EditGroupRolesPropsInterface> = (
                                 <RenderChip 
                                     { ...getTagProps({ index }) }
                                     key={ index }
-                                    primaryText={ option.display }
+                                    primaryText={ "" }
```
Comment (Achintha444) [inline apps/console/src/features/groups/components/edit-group/edit-group-roles.tsx:121]: Remove `primaryText` prop.

File: apps/console/src/features/groups/components/group-common-components/render-chip.tsx (hunk 2 above)
Comment (Achintha444) [reply apps/console/src/features/groups/components/group-common-components/render-chip.tsx:63]: +1

File: apps/console/src/features/groups/components/group-common-components/render-chip.tsx [hunk 4]
```diff
@@ -19,7 +19,6 @@
 import Chip, { ChipProps } from "@oxygen-ui/react/Chip";
 import { IdentifiableComponentInterface, RolesMemberInterface } from "@wso2is/core/models";
 import React, { FunctionComponent, ReactElement, SyntheticEvent, useState } from "react";
-import { ChipMoreDetails } from "./chip-more-details";
```
Comment (Achintha444) [inline apps/console/src/features/groups/components/group-common-components/render-chip.tsx:22]: Remove `chip-more-details` file as well.

File: apps/console/src/features/groups/components/group-common-components/render-chip.tsx (hunk 1 above)
Comment (Shenali-SJ) [reply apps/console/src/features/groups/components/group-common-components/render-chip.tsx:33]: Fixed with 8cb8bf2176f03d3406c2b5b4530b56bf77c07a78

File: apps/console/src/features/groups/components/group-common-components/render-chip.tsx (hunk 2 above)
Comment (Shenali-SJ) [reply apps/console/src/features/groups/components/group-common-components/render-chip.tsx]: Fixed with 8cb8bf2176f03d3406c2b5b4530b56bf77c07a78

File: apps/console/src/features/groups/components/edit-group/edit-group-roles.tsx (hunk 3 above)
Comment (Shenali-SJ) [reply apps/console/src/features/groups/components/edit-group/edit-group-roles.tsx]: Fixed with 8cb8bf2176f03d3406c2b5b4530b56bf77c07a78

File: apps/console/src/features/groups/components/group-common-components/render-chip.tsx (hunk 4 above)
Comment (Shenali-SJ) [reply apps/console/src/features/groups/components/group-common-components/render-chip.tsx:22]: Fixed with 8cb8bf2176f03d3406c2b5b4530b56bf77c07a78

**Comment by Shenali-SJ** [comment]
The following screenshot displays how the roles will be shown when there are multiple org audience roles.
image

**Review by Achintha444 — APPROVED** [review]

**Pull request merged**